spec/jmap/push.mdown

```diff
@@ -1,69 +1,94 @@
 
-## Web hook
+- **url**: `String`
+  An absolute URL where the JMAP server will POST the data for the push message.
+  This MUST begin with `https://`.
+- **expires**: `Date|null`
+  The time this push subscription expires. If specified, the JMAP server MUST NOT make further requests to this resource after this time. It MAY automatically remove the push subscription at or after this time.
+- **keys**: `Object|null`
+  Clients generated encryption keys. If supplied the server MUST use them as specified in <https://tools.ietf.org/html/draft-ietf-webpush-encryption-09> to encrypt all data sent to the push subscription. The object MUST have the following properties:
 
-The second push mechanism is to register a callback URL to which the JMAP server will make an HTTPS POST request whenever the event occurs. The request MUST have a content type of `application/json` and contain the same UTF-8 JSON encoded object as described above as the body.
+    - **p256dh**: the P-256 ECDH Diffie-Hellman public key as described in <https://tools.ietf.org/html/draft-ietf-webpush-encryption-09>, encoded in URL-safe base64 representation as defined in [@!RFC4648].
+    - **auth**: the authentication secret as described in <https://tools.ietf.org/html/draft-ietf-webpush-encryption-09>, encoded in URL-safe base64 representation as defined in [@!RFC4648].
 
-The JMAP server MUST also set the following headers in the POST request:
-- `X-JMAP-EventType: state`
-- `X-JMAP-User: ${username}` where `${username}` is the username of the authenticated user for which the push event occurred.
+Clients may register the push subscription with the JMAP server, which will then make a POST request to the associated push endpoint whenever an event occurs.
 
-The JMAP server MUST be capable of following at least one redirect, but
-MAY choose not to do so for policy reasons. If the final response code from the server is `2xx`, the callback is considered a success. If the response code is `503` (Service Unavailable), the JMAP server MAY try again later (but may also just drop the event). If the response code is `429` (Too Many Requests) the JMAP server SHOULD attempt to reduce the frequency of pushes to that URL. Any other response code MUST be considered a **permanent failure** and the callback should be deregistered (not tried again even for future events unless explicitly re-registered by the client).
+The POST request MUST have a content type of `application/json` and contain the utf-8 JSON encoded *StateChange* object as the body. The request MUST have a `TTL` header, and MAY have `Urgency` and/or `Topic` headers, as specified in section 5 of [@!RFC8030].
 
-The URL set by the client MUST use the HTTPS protocol and SHOULD encode within it a unique token that can be verified by the server to know that the request comes from the JMAP server the authenticated client connected to.
+The JMAP server MUST follow at least one redirect, and MAY follow more.
```

Comment:
This seems overly prescriptive.  Are you talking about profiling HTTP?

Reply:
No, that whole line is unnecessary; I'll cut it.